### ISSUE TITLE
Issue with the Current Version of ND4J

### ISSUE THREAD

**Issue by man2code** [deeplearning4j/nd4j#518]
I was using the  0.0.3.5.5 version of nd4j with an in-house MLP and everything was working fine. But when I updated it to the 0.4-rc3.6 version, the same MLP classified all the instances in the same class.
I got the same issue running the MLP in dl4j with the same nd4j version.
BTW, the newer nd4j version seems to be slower.

**Comment by raver119** [comment]
Current version is 0.4-rc3.7

**Comment by man2code** [comment]
Right but the same issue occurs in 0.4-rc3.7.

**Comment by agibsonccc** [comment]
Version of java? "Seems to be" and "I ran this with actual numbers" are 2 different responses that would be great to get. You haven't actually described a problem yet. DL4j has ALOT of knobs for tuning. "In house neural net" doesn't give me code, configurations or anything. Please give actual detail in the issue or I'm closing this.

Thanks.

**Comment by man2code** [comment]
Two Java versions:
java version "1.8.0_45"
Java(TM) SE Runtime Environment (build 1.8.0_45-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.45-b02, mixed mode)

java version "1.7.0_55"
OpenJDK Runtime Environment (IcedTea 2.4.8) (suse-8.36.1-x86_64)
OpenJDK 64-Bit Server VM (build 24.55-b03, mixed mode)

-----

I believe you didn't carefully read what I wrote.
With an in-house MLP implementation (not DL4J) +  ND4J 0.0.3.5.5, things worked fine. 
With the SAME MLP implementation/configuration + ND4J 0.4-rc3.7 or 0.4-rc3.6, the problem happens. 
So, I believe the problem is in the ND4J newer versions.
I understand that MLP has many tuning parameters. Believe me, I've tried 
many possible values for them and nothing has changed.

Regarding the speed, I was just trying to help saying about my experience with the 
newer versions.
